app/Http/Controllers/Group/WorkspaceController.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace App\Http\Controllers\Group;
+
+use App\Http\Controllers\Controller;
+use App\Models\Bookmark;
+use App\Models\Group;
+use App\Models\LegiScan\Session;
+use App\Models\Workspace;
+
+class WorkspaceController extends Controller
+{
+    public function show(Group $group, Workspace $workspace)
+    {
+        $sessionYears = Session::validYears();
```

Comment:
I get why you did this. However. This is now a testing blockade, because the method is `static`. This dependency cannot be mocked and registered in the container. It cannot be avoided.

I _think_ you can get away with it here, sort of, but the much better think to do is to make a third class that has the single responsibility of assembling this query, and then pick one of:

- `app(ValidSessions::class)->getValidYears()`
- Create a Facade to wrap that class in and call `ValidSessions::validYears()`

Either way, because you have an instantiable class with a non-static method, you can mock and insert it into the service container and avoid a blockade.

Again, you're probably fine for the moment and can refactor this later. However. More troubling to my OCD:

- Classes & Properties are Nouns
- Methods are Verbs
- "Valid Years" is a Noun. Please change it to be some kind of verb.

Reply:
What about `SessionHelper::getValidYears()`? I think I remember you are not a fan of the word "helper".

Reply:
Fixed